### DISCUSSION TITLE
ctx.assert(false, 401, "foo") ===  ctx.throw("foo", 401) ?

### DISCUSSION THREAD

**Discussion by SimonAM** [koajs/koa#1674] [Q&A] [answered]
Are ctx.assert and ctx.throw interchangable so if i have to run multiple side effects in an if-statement i don't need to run assert inside? 

The docs says assert is "similar to .throw()", and i can see they return the same type. But there could still be other side effects. 

If so, could the docs be updated with: "assert returns the same as .throw() if the first argument is false".

**Comment by miwnwski** [comment, marked as answer]
Assert only throws an exception if the statement is falsy, while throw always throws an exception on call. These concepts are translated from traditional assert and throw semantics. If you still feel these are unclear, maybe PR a change and people can review it? Subjectively I don't think it's unclear, because I assume the purpose of assert and throw is commonly known, but maybe I'm wrong or misunderstand the issue?
